### PR TITLE
Make astropy.tests.helper part of the public API

### PR DESCRIPTION
**Pull request by eteq**
This grew out of a discussion with @cdeil as part of one of the GSoC projects.  It turns out we aren't currently documenting the functions and markers in `astropy.tests.helper`.  This means e.g. `remote_data` or `quantity_allclose` can't be relied on or understood by affiliated package writers as its not part of the public API.  Some of it is obliquely referenced in the testing guidelines, but not directly documented.

This PR rectifies that by adding `astropy.tests.helper` documentation to the astropy docs, thereby making it part of the public API.  One weird quirk: it turns out `numpy` is not doing intersphinx correctly, so `numpy.testing` doesn't link up to the right numpy doc page.  I worked around this by just explicitly providing a hyperlink, bypassing the intersphinx machinery.

### CODE REVIEW

**Comment by cdeil** [comment]
:+1:  Thanks!

> numpy is not doing intersphinx correctly, so numpy.testing doesn't link up to the right numpy doc page

File a numpy issue?

File: CHANGES.rst [hunk 1]
```diff
@@ -174,6 +174,10 @@ New Features
 
   - Enhanced text representation of ``WCS`` objects. [#3604]
 
+- The `astropy.tests.helper` module is now part of the public API (and has a
```
Comment (embray) [inline CHANGES.rst]: Minor, annoying nitpick but this should use double-backticks.  The (current) changelog format does not assume any attempt to resolve API doc references (I'm hoping to address that in the new format eventually...)

File: docs/testhelpers.rst [hunk 2]
```diff
@@ -0,0 +1,39 @@
+.. _testhelpers:
+
+************************************************
+Astropy Testing helpers (`astropy.tests.helper`)
+************************************************
+
+To easy development of tests that work with Astropy, the
```
Comment (embray) [inline docs/testhelpers.rst]: "easy" -> "ease" ?

File: docs/testhelpers.rst [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+`astropy.tests.helper` module provides some utility functions to make
+test that use astropy conventions or classes easier to work with. E.g.,
```
Comment (embray) [inline docs/testhelpers.rst]: "test" -> "tests", "astropy" -> "Astropy"

Also "with. E.g.," -> "with, e.g."

File: docs/testhelpers.rst [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+functions to test for near-equality of `~astropy.units.Quantity` objects.
+
+The functionality here is not exhaustive, because much of the useful
+tools are either in the standard library, py.test, or
+`numpy.testing <http://docs.scipy.org/doc/numpy/reference/routines.testing.html>`_.
+This module contains primarily functionality specific to astropy or
```
Comment (embray) [inline docs/testhelpers.rst]: "astropy" -> "Astropy"

File: docs/testhelpers.rst [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+    from astropy.tests.helper import remote_data
+
+    @remote_data
+    def test_something():
+        """
+        This test dowloads something from the intenet
+        """
+        # test code here ...
+
```
Comment (embray) [inline docs/testhelpers.rst]: Maybe worth adding a note here that tests marked with `@remote_data` can be enabled/disabled using the `--remote-data` argument to `setup.py test`

**Comment by embray** [comment]
Made a few nitpicks, but otherwise :+1:  from me.

File: CHANGES.rst (hunk 1 above)
Comment (eteq) [inline CHANGES.rst]: Oops, yeah, I know about that in theory but seem to forget almost every time out of habit.  Will fix.

**Comment by eteq** [comment]
Alright, I think I addressed all of @embray's comments. So I think this is good to go once the tests pass.

**Comment by eteq** [comment]
Oh, and @cdeil - there is an issue for it *somewhere* on numpy, or there was at some point (I know I found it via googling...).  Having trouble finding it now, though...

File: CHANGES.rst (hunk 1 above)
Comment (embray) [inline CHANGES.rst]: Yeah, it's an annoying little thing.

**Comment by cdeil** [comment]
It worked: http://astropy.readthedocs.org/en/latest/testhelpers.html?highlight=remote_data

Thanks @eteq!

**Pull request merged**